tests/unit/test_api_urls.py

```diff
@@ -160,37 +160,89 @@ def test_api_search_wrong_parameter(self):
             data = {'state': 'closed', 'milestone': 2}
             patch_data = json.dumps(data)
             rv = self.app.patch('/api/issues/1/edit', data=patch_data,
                                 environ_base=headers)
             self.assertEqual(rv.status_code, 403)
 
-    def test_api_patch_issue_too_many_json_elements(self):
+    @patch('webcompat.db.User.query')
+    def test_api_patch_issue_too_many_json_elements(self, db_user_mock):
         """Patching the issue - Too many elements in the JSON."""
         with webcompat.app.app_context():
+
+            # mock authenticated user
+            db_user_mock.return_value.get.return_value = get_user_mock()
+            with self.app.session_transaction() as sess:
+                sess['user_id'] = 1
+
             data = {'state': 'open', 'milestone': 2, 'foobar': 'z'}
             patch_data = json.dumps(data)
             rv = self.app.patch('/api/issues/1/edit', data=patch_data,
                                 environ_base=headers)
             self.assertEqual(rv.status_code, 403)
 
+    def test_api_patch_issue_without_auth(self):
+        """Patching the issue - Request without auth."""
+        with webcompat.app.app_context():
+            data = {'state': 'open', 'milestone': 2}
+            patch_data = json.dumps(data)
+            rv = self.app.patch('/api/issues/1/edit', data=patch_data,
+                                environ_base=headers)
+            self.assertEqual(rv.status_code, 403)
+
+    @patch('webcompat.db.User.query')
     @patch('webcompat.api.endpoints.api_request')
-    def test_api_patch_issue_valid_request(self, github_data):
+    def test_api_patch_issue_valid_request(self, github_data, db_user_mock):
         """Patching the issue - Valid request."""
         with webcompat.app.app_context():
-            github_data.return_value = mock_api_response(
-                {'status_code': 200, 'content': '[]'})
+
+            # mock authenticated user
+            db_user_mock.return_value.get.return_value = get_user_mock()
+            with self.app.session_transaction() as sess:
+                sess['user_id'] = 1
+
+            github_data.return_value = ('[]', 200, {})
             data = {'state': 'open', 'milestone': 2}
             patch_data = json.dumps(data)
             rv = self.app.patch('/api/issues/1/edit', data=patch_data,
                                 environ_base=headers)
             self.assertEqual(rv.status_code, 200)
 
+    @patch('webcompat.db.User.query')
+    @patch('webcompat.api.endpoints.api_request')
+    def test_api_set_labels_valid_request(self, github_data, db_user_mock):
+        """Setting labels - Valid request."""
+        with webcompat.app.app_context():
+
+            # mock authenticated user
+            db_user_mock.return_value.get.return_value = get_user_mock()
+            with self.app.session_transaction() as sess:
+                sess['user_id'] = 1
+
```

Comment:
Tried mocking `g.user` as the documentation suggests in https://flask.palletsprojects.com/en/2.0.x/testing/#faking-resources-and-context , but it didn't work for some reason :/ 

So here I'm mocking session and `User.query.get` and `g.user` gets assigned a mocked value in `before_request`
https://github.com/webcompat/webcompat.com/blob/main/webcompat/views.py#L65. Maybe there is a better way.

Reply:
```
Flask==1.1.2
```

and i double check and the doc is similar for 1.1.2
https://flask.palletsprojects.com/en/1.1.x/testing/#faking-resources-and-context